### ISSUE TITLE
Ensure all SQL writes are in try/catch

### ISSUE THREAD

**Issue by gfitzp** [gfitzp/fitzflix#72]
```
Traceback (most recent call last):
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1203, in _execute_context
   conn = self._revalidate_connection()
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 472, in _revalidate_connection
   raise exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: Can't reconnect until invalid transaction is rolled back
```

**Comment by gfitzp** [comment]
Fixed in the videos.py file in ddc478d5bacf957d36723289bc75ef2a783189dc, need to check all other commits before closing.

**Comment by gfitzp** [comment]
Still getting timeouts.

```
Traceback (most recent call last):
  File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/connections.py", line 705, in _write_bytes
    self._sock.sendall(data)
BrokenPipeError: [Errno 32] Broken pipe

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
    self.dialect.do_execute(
  File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)
  File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/cursors.py", line 163, in execute
    result = self._query(query)
  File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/cursors.py", line 321, in _query
    conn.query(q)
  File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/connections.py", line 504, in query
    self._execute_command(COMMAND.COM_QUERY, sql)
  File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/connections.py", line 763, in _execute_command
    self._write_bytes(packet)
  File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/connections.py", line 708, in _write_bytes
    raise err.OperationalError(
pymysql.err.OperationalError: (2006, "MySQL server has gone away (BrokenPipeError(32, 'Broken pipe'))")
```

```
Traceback (most recent call last):
  File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1203, in _execute_context
    conn = self._revalidate_connection()
  File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 472, in _revalidate_connection
    raise exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: Can't reconnect until invalid transaction is rolled back
```

Attempting changes to connection pool in 7af5d8d8e899b442d7385238b99c18f0ebb7772c to see if that helps. See https://medium.com/@heyjcmc/controlling-the-flask-sqlalchemy-engine-a0f8fae15b47 and http://www.ilian.io/sqlalchemy-and-lost-connection-mysql-server-during-query/

**Comment by gfitzp** [comment]
Still having issues.

```
Traceback (most recent call last):
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/connections.py", line 705, in _write_bytes
   self._sock.sendall(data)
BrokenPipeError: [Errno 32] Broken pipe

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
   self.dialect.do_execute(
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
   cursor.execute(statement, parameters)
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/cursors.py", line 163, in execute
   result = self._query(query)
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/cursors.py", line 321, in _query
   conn.query(q)
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/connections.py", line 504, in query
   self._execute_command(COMMAND.COM_QUERY, sql)
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/connections.py", line 763, in _execute_command
   self._write_bytes(packet)
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/pymysql/connections.py", line 708, in _write_bytes
   raise err.OperationalError(
pymysql.err.OperationalError: (2006, "MySQL server has gone away (BrokenPipeError(32, 'Broken pipe'))")
```

```
Traceback (most recent call last):
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1203, in _execute_context
   conn = self._revalidate_connection()
 File "/Users/gfitzp/Sites/fitzflix/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 472, in _revalidate_connection
   raise exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: Can't reconnect until invalid transaction is rolled back
```

**Comment by gfitzp** [comment]
Changing server in connect string from `localhost` to `127.0.0.1`.

**Comment by gfitzp** [comment]
Still getting the issue, I think now it's because the database connection is outside of the Flask request; see https://stackoverflow.com/a/28575225

**Comment by gfitzp** [comment]
Appears to be fixed in 62c846a994673c4117c610ec79d715d5759bb706

**Comment by gfitzp** [comment]
```
2023-07-09 10:14:36 13450 [Warning] Aborted connection 13450 to db: 'fitzflix_db' user: 'fitzflix_user' host: 'localhost' (Got an error reading communication packets)
```

https://severalnines.com/blog/common-mysql-error-got-error-reading-communication-packet/

Use `mysqld --help --verbose` for checking configuration settings

**Comment by gfitzp** [comment]
Removed DB connection pooling in 9eb2665

**Comment by gfitzp** [comment]
Even without connection pooling got error:

```
sqlalchemy.exc.OperationalError: (pymysql.err.OperationalError) (2006, "MySQL server has gone away (BrokenPipeError(32, 'Broken pipe'))")
[SQL: SELECT [file.id](http://file.id/) AS file_id, file.untouched_basename AS file_untouched_basename, file.media_library AS file_media_library, file.file_path AS file_file_path, file.dirname AS file_dirname, file.basename AS file_basename, file.plex_title AS file_plex_title, file.movie_id AS file_movie_id, file.feature_type_id AS file_feature_type_id, file.series_id AS file_series_id, file.season AS file_season, file.episode AS file_episode, file.last_episode AS file_last_episode, file.version AS file_version, file.quality_id AS file_quality_id, file.fullscreen AS file_fullscreen, file.crop AS file_crop, file.container AS file_container, file.format AS file_format, file.codec AS file_codec, file.video_bitrate_kbps AS file_video_bitrate_kbps, file.date_added AS file_date_added, file.date_updated AS file_date_updated, file.date_localized AS file_date_localized, file.date_transcoded AS file_date_transcoded, file.date_archived AS file_date_archived, file.aws_untouched_key AS file_aws_untouched_key, file.aws_untouched_date_uploaded AS file_aws_untouched_date_uploaded, file.aws_untouched_date_deleted AS file_aws_untouched_date_deleted 
FROM file]
```

```
sqlalchemy.exc.PendingRollbackError: Can't reconnect until invalid transaction is rolled back. (Background on this error at: https://sqlalche.me/e/14/8s2b)
```

**Comment by gfitzp** [comment]
Might be fixed in 2449306 ?

**Comment by gfitzp** [comment]
Issue hasn't occurred in a while, closing ticket 🤞🏻
